### PR TITLE
Fork/bounds checking

### PR DESCRIPTION
**Pull request by ryco117**
##### Description
Applied checking that the Ethereum-network matches between the browser and bounty for fulfilling, increasing, killing, and processing bounties. Added check to ensure an owner of a bounty cannot fulfill one in their possession. Added deadline and stage check for fulfilling bounty as well as stage check for processing bounty. Also factored out a shared function `string web3NetworkIdToString(EthNetId)` for easy retrieving of a pretty name for current given network id and a shared enum for stages.

##### Checklist
- [x] linter status: 100% pass
- [x] changes don't break existing behavior
- [x] commit message follows [commit guidelines](https://github.com/gitcoinco/web/blob/master/docs/CONTRIBUTING.md#step-4-commit)

##### Affected core subsystem(s)

js/pages

##### Testing
Manually created, fulfilled, increasing, and processed bounties on my local copy (in addition to testing that the new boundaries were enforced.)

##### Refers/Fixes

Fixes: #445

### CODE REVIEW

File: app/assets/v2/js/pages/increase_bounty.js [hunk 1]
```diff
@@ -105,53 +105,78 @@ $(document).ready(function() {
...
-          value: 0,
-          gasPrice: web3.toHex($('#gasPrice').val()) * Math.pow(10, 9)
-        },
-        approveSuccessCallback
-      );
-    }
+      var bountyNetwork = result['network'];
+
+      // Determine if browser and bounty networks match
+      web3.version.getNetwork((error, netId) => {
```
Comment (owocki) [inline app/assets/v2/js/pages/increase_bounty.js]: i wonder if theres a way to DRY for this check across multiple of these JS files

Comment (ryco117) [reply app/assets/v2/js/pages/increase_bounty.js]: Sorry, not sure I follow.. Are you asking if it can be better factored out into a shared function?

Comment (owocki) [reply app/assets/v2/js/pages/increase_bounty.js]: yes. DRY = dont repeat yourself

File: app/assets/v2/js/pages/increase_bounty.js [hunk 2]
```diff
@@ -105,53 +105,78 @@ $(document).ready(function() {
...
-          value: ethAmount,
-          gasPrice: web3.toHex($('#gasPrice').val()) * Math.pow(10, 9)
-        },
-        web3Callback
-      );
-    }
+    // Use DB to determine which networks each bounty is on.
+    var uri = '/api/v0.1/bounties/?github_url=' + issueURL;
+
+    $.get(uri, function(results, status) {
```
Comment (owocki) [inline app/assets/v2/js/pages/increase_bounty.js]: sometimes were using the API as the source of truth and sometimes we're using the smart contract.. is there a rule you followed to decide which to use where?

File: app/assets/v2/js/shared.js [hunk 3]
```diff
@@ -636,6 +636,31 @@ var trigger_form_hooks = function() {
   trigger_faucet_form_web3_hooks();
 };
 
+var web3NetworkIdToString = function(netId) {
```
Comment (owocki) [inline app/assets/v2/js/shared.js]: nice!

File: app/assets/v2/js/pages/increase_bounty.js [hunk 4]
```diff
@@ -105,53 +105,78 @@ $(document).ready(function() {
...
+        var errormsg = undefined;
+
+        if (bountyAmount == 0 || open == false) {
+          errormsg = 'No active funded issue found at this address. Are you sure this is an active funded issue?';
+        }
+        if (fromAddress != web3.eth.coinbase) {
+          errormsg = 'Only the address that submitted this funded issue may increase the payout.';
+        }
+        if (browserNetwork != bountyNetwork) {
+          errormsg = 'The browser must be connected to same Ethereum network that the bounty was deployed to.';
```
Comment (owocki) [inline app/assets/v2/js/pages/increase_bounty.js]: might be nice to let the user know which network that is in the err message

File: app/assets/v2/js/pages/increase_bounty.js (hunk 2 above)
Comment (ryco117) [reply app/assets/v2/js/pages/increase_bounty.js]: In this case, have to use the DB to resolve which of the two blockchains the issue is backed on. I only used the blockchain as truth for expiration-date and status.

Comment (owocki) [reply app/assets/v2/js/pages/increase_bounty.js]: ok makes sense

**Comment by owocki** [comment]
mind resolving the conflicst here?

**Comment by mbeacom** [comment]
This branch appears to have a bad rebase.  I recommend rebuilding the branch from `upstream/master` if possible.  Might need to cherrypick.

**Comment by abitrolly** [comment]
@ryco117 do you need help with rebasing?

**Comment by ryco117** [comment]
Sorry, I've been busy lately. I'll merge with master and commit soon,
shouldn't have too many conflicts.

**Comment by ryco117** [comment]
Kinda screwed up this fork, moved changes (+ recommended improvements) over to PR #870

**Pull request closed without merging**